File: C/Algorithms/121.c
#define MAX(a, b) a >= b ? a : b

int maxProfit(int *prices, int pricesSize) {
  int max_price = 0, min_price = prices[0];
  for (int i = 0; i < pricesSize; i++) {
    int price = prices[i];
    if (price <= min_price)
      min_price = price;
    else {
      int diff = price - min_price;
      max_price = MAX(max_price, diff);
    }
  }
  return max_price;
}

File: Javascript/Algorithms/121.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
  const max = Math.max;
  let max_price = 0, min_price = prices[0];
  for (let price of prices) {
    if (price <= min_price)
      min_price = price;
    else
      max_price = max(max_price, price - min_price);
  }
  return max_price;
};